### DISCUSSION TITLE
AttributeError: 'Flow' object has no attribute 'terminal_state_handler'

### DISCUSSION THREAD

**Discussion by heiqs** [PrefectHQ/prefect#4804] [Q&A]
I see the following log when running a job. The task is successful but I get the following error:

`09:32:17 INFO CloudFlowRunner Flow run SUCCESS: all reference tasks succeeded 09:32:17 ERROR CloudFlowRunner Unexpected error: AttributeError("'Flow' object has no attribute 'terminal_state_handler'") Traceback (most recent call last):   File "/usr/local/lib/python3.8/site-packages/prefect/engine/runner.py", line 48, in inner     new_state = method(self, state, *args, **kwargs)   File "/usr/local/lib/python3.8/site-packages/prefect/engine/flow_runner.py", line 669, in get_flow_run_state     state = self.determine_final_state(   File "/usr/local/lib/python3.8/site-packages/prefect/engine/flow_runner.py", line 720, in determine_final_state     if self.flow.terminal_state_handler: AttributeError: 'Flow' object has no attribute 'terminal_state_handler'`

**Comment by zanieb** [comment]
Hey @heiqs -- this is typically caused by a version mismatch between your flow and runtime environment. Your flow was pickled with an old version that does not have a `terminal_state_handler` but the version of prefect executing the flow is trying to access it.
